test/unit/user_test.rb

```diff
@@ -220,4 +221,27 @@ class UserTest < ActiveSupport::TestCase
     #as the username as "jeff" exists, hence username = "jeff" + 2 digit alphanumeric code will be created
     assert_not_equal jeffrey.username, "jeff"
   end
+
+  test 'generate token and validate token for user email verification' do
```

Comment:
Almost there, just split these all these tests into multiple functions. 

In unit tests, you should try tests to be as small as possible so that they test one small functionality properly. When they fail, developer easily get to know which test failed.